walk/walk.go

```diff
@@ -356,3 +359,42 @@ func resolveFileInfo(wc *walkConfig, dir, rel string, ent fs.DirEntry) fs.DirEnt
 	}
 	return fs.FileInfoToDirEntry(fi)
 }
+
+func buildTrie(c *config.Config, isIgnored isIgnoredFunc) (*pathTrie, error) {
+	trie := &pathTrie{}
+
+	eg := errgroup.Group{}
+	eg.SetLimit(100)
+	eg.Go(func() error {
+		return walkDir(c.RepoRoot, "", &eg, isIgnored, trie)
+	})
+
+	return trie, eg.Wait()
+}
+
+// walkDir recursively descends path, calling walkDirFn.
+func walkDir(root, rel string, eg *errgroup.Group, isIgnored isIgnoredFunc, trie *pathTrie) error {
+	dirs, err := os.ReadDir(path.Join(root, rel))
+	if err != nil {
+		return err
+	}
+
+	for _, d1 := range dirs {
```

Comment:
What's up with the names? Should we do dir, dirName, dirPath?

Reply:
They're maybe not dirs, it should probably be ent, entName, entPath or something